packages/nimble-components/src/table/index.ts

```diff
@@ -343,28 +392,81 @@ export class Table<
 
     private refreshRows(): void {
         const rows = this.table.getRowModel().rows;
-        this.tableData = rows.map(row => {
+        this.tableData = rows.map((row, i) => {
             const rowState: TableRowState<TData> = {
                 record: row.original,
-                id: row.id
+                id: row.id,
+                isGrouped: row.getIsGrouped(),
+                isExpanded: row.getIsExpanded(),
+                groupRowValue: row.getIsGrouped()
+                    ? row?.getValue(row.groupingColumnId!)
+                    : undefined,
+                nestingLevel: row.depth,
+                leafItemCount: row
+                    .getLeafRows()
+                    .filter(leafRow => leafRow.getLeafRows().length === 0)
+                    .length,
+                groupColumn: this.getGroupRowColumn(i)
             };
             return rowState;
         });
         this.virtualizer.dataChanged();
     }
 
+    private getGroupRowColumn(rowIndex: number): TableColumn | undefined {
+        const groupedId = this.table.getRowModel()!.rows[rowIndex]?.groupingColumnId;
+        if (groupedId !== undefined) {
+            return this.columns.find(c => c.internalUniqueId === groupedId);
+        }
+
+        return undefined;
+    }
+
     private updateTableOptions(
-        updatedOptions: Partial<TanStackTableOptionsResolved<TData>>
+        updatedOptions: Partial<TanStackTableOptionsResolved<TData>>,
+        skipRowRefresh?: boolean
```

Comment:
Do you still need to modify this function to have `skipRowRefresh`? I don't see that used anywhere.